File: backend/team.py
from pydantic import BaseModel
from typing import Optional

class Team(BaseModel):
    team_id: Optional[int] = None
    name: str
    invite_code: str
    approved: bool

    #def __init__(self, team_id: int, name: str, invite_code: str, approved: bool):
    #    self.team_id = team_id
    #    self.name = name
    #    self.invite_code = invite_code
    #    self.approved = approved

File: frontend/src/app/model/team.ts
import {SafeUrl} from "@angular/platform-browser";
import {Person} from "./person";

export interface Team {
  team_id: number;
  name: string;
  invite_code: string;
  approved: boolean;
  QRCode: SafeUrl;
  inviteLink: string;
  teamMax: number;
  members: Person[];
}
